src/peft/tuners/lora/layer.py

```diff
@@ -150,6 +153,30 @@ def reset_lora_parameters(self, adapter_name, init_lora_weights):
             nn.init.zeros_(self.lora_embedding_A[adapter_name])
             nn.init.normal_(self.lora_embedding_B[adapter_name])
 
+    def olora_init(self, adapter_name):
+        dtype = self.base_layer.weight.dtype
+        if dtype in [torch.int8, torch.uint8]:
+            warnings.warn("You are using OLoRA with quantized weights. This is still experimental.")
+            base_layer_copy = deepcopy(self.base_layer)  # Make FSDP happy
+            weight_tensor = dequantize_module_weight(base_layer_copy)
```

Comment:
Experimental quantization support. I am not familiar with how peft internally dequantizes so I came up with this. Maybe you can help me out on this one. It works ok with `load_in_4bit` and `load_in_8bit` for now.